libvast/src/transform_steps/select.cpp

```diff
@@ -39,19 +40,36 @@ select_step::select_step(std::string expr) : expression_(caf::no_error) {
   }
 }
 
-caf::expected<table_slice> select_step::operator()(table_slice&& slice) const {
-  if (!expression_)
+caf::error select_step::add(vast::id offset, type layout,
+                            std::shared_ptr<arrow::RecordBatch> batch) {
+  VAST_DEBUG("select_step add");
+  if (!expression_) {
+    transformed_.clear();
     return expression_.error();
-  auto tailored_expr = tailor(*expression_, slice.layout());
-  if (!tailored_expr)
+  }
+  auto tailored_expr = tailor(*expression_, layout);
+  if (!tailored_expr) {
+    transformed_.clear();
     return tailored_expr.error();
-  auto new_slice = filter(slice, *tailored_expr);
-  if (new_slice)
-    return *new_slice;
+  }
+  auto new_slice
+    = filter(arrow_table_slice_builder::create(batch, layout), *tailored_expr);
+  if (new_slice) {
+    auto as_batch = as_record_batch(
+      *new_slice); // FIXME: ask: keepalive batch until create is called()
```

Comment:
Please take care of the FIXME

Reply:
There is a separate story to handle this issue: [SC-31007](https://app.shortcut.com/tenzir/story/31007/implement-proper-lifetime-extension-for-record-batches-created-from-arrow-table-slices).